typify-impl/src/value.rs

```diff
@@ -162,12 +163,21 @@ impl TypeEntry {
                 if !value.is_number() {
                     return None;
                 }
-                let val = match proc_macro2::Literal::from_str(&format!("{}_{}", value, type_name))
-                {
-                    Ok(v) => v,
-                    Err(_) => unreachable!(),
-                };
-                TokenStream::from(proc_macro2::TokenTree::from(val))
+                if type_name.starts_with(STD_NUM_NONZERO_PREFIX) {
+                    let type_path = syn::parse_str::<syn::TypePath>(type_name).unwrap();
+                    let num = proc_macro2::Literal::from_str(value.to_string().as_str()).unwrap();
+
+                    quote! {
+                        #type_path::new(#num).unwrap()
```

Comment:
Not a huge deal but, if you've already validated the `value`, then you could use `new_unchecked()` here to avoid the unwrap. I can't tell if you've validated it though.

Reply:
I considered that, but didn't want to generate

```rust
    unsafe { #type_path::new_unchecked(#num) }
```